operation/manager_test.go

```diff
@@ -11,155 +11,167 @@ import (
 	"go.viam.com/test"
 )
 
-func TestSingleOperationManager(t *testing.T) {
-	ctx := context.Background()
-	som := SingleOperationManager{}
+var (
+	ctx = context.Background()
+	som = SingleOperationManager{
```

Comment:
If I'm reading this correctly, we've now got global `ctx` and `som` variables that get reused in all the tests. The worry with that is that if one test fails and leaves these in a bad state, other tests will start failing due to side effects bleeding over from the earlier tests. 

What I meant by my previous comment about avoiding boilerplate is that in other languages I'm familiar with, you can have a setup function that constructs new copies of the state that all tests want, and I suspect Go probably has something similar but I don't actually know about this myself. For example, here's roughly what it looks like in Python:

```python
import pytest
class FooTester(pytest.TestCase):
    def setup(self):
        self.ctx = context.Background()  # Constructs a brand new context for each test function
        self.som = SingleOperationManager(...)  # Constructs a brand new SOM for each test function

    def testThing1(self):
        self.assertFalse(self.som.hasRunningOperation())
        self.assertEmpty(self.ctx)

    def testThing2(self):
        ...
```

`self.ctx` and `self.som` are completely different, freshly created objects in `testThing1` and `testThing2`, so that the two tests don't interfere with each other, but you only needed to define the objects once in `setup`. Does Go's test framework have something analogous to this?

Reply:
Hmmm. In this case we would have to set up only an empty `SingleOperationManager` for this file. The contexts are either unused, set up as `context.Backgrounds()` using the `som.New` method.

So I'll just add the boiler plate back for each test. We weren't doing that anyway for the older file and that was a main cause of the flakiness.

Reply:
I've glanced through https://pkg.go.dev/testing and https://go.dev/src/testing/testing.go, and it indeed looks like there is no setup-like function in Go's test infrastructure. Sorry to ask for the impossible!